src/EnergyPlus/PlantLoopHeatPumpEIR.cc

```diff
@@ -387,12 +450,22 @@ void EIRPlantLoopHeatPump::calcAvailableCapacity(EnergyPlusData &state, Real64 c
     // evaluate capacity modifier curve and determine load side heat transfer
     // any adjustment to outlet water temp set point requires some form of iteration
     for (int loop = 0; loop < 2; ++loop) {
-        capacityModifierFuncTemp = Curve::CurveValue(state, this->capFuncTempCurveIndex, loadSideOutletSetpointTemp, this->sourceSideInletTemp);
-
-        availableCapacity = this->referenceCapacity * capacityModifierFuncTemp;
 
-        // apply air source HP dry air heating capacity correction
-        availableCapacity *= heatingCapacityModifierASHP(state);
+        if (this->heatRecoveryIsActive) {
+            if (this->heatRecoveryCapFTempCurveIndex > 0) {
+                capacityModifierFuncTemp =
+                    Curve::CurveValue(state, this->heatRecoveryCapFTempCurveIndex, loadSideOutletSetpointTemp, this->heatRecoveryInletTemp);
+            } else {
+                capacityModifierFuncTemp =
+                    Curve::CurveValue(state, this->capFuncTempCurveIndex, loadSideOutletSetpointTemp, this->heatRecoveryInletTemp);
+            }
+            availableCapacity = this->referenceCapacity * capacityModifierFuncTemp;
+        } else {
+            capacityModifierFuncTemp = Curve::CurveValue(state, this->capFuncTempCurveIndex, loadSideOutletSetpointTemp, this->sourceSideInletTemp);
+            availableCapacity = this->referenceCapacity * capacityModifierFuncTemp;
+            // apply air source HP dry air heating capacity correction
+            availableCapacity *= heatingCapacityModifierASHP(state);
+        }
```

Comment:
If the heat recovery is off or not specified, then the air-source capacity modifier curve is used.